src/types.ts

```diff
@@ -2,150 +2,86 @@
-}
-
-export enum Currency {
-    USD = "usd",
-    EUR = "eur"
+  FULL = 0,
+  PARTIAL = 1
 }
 
 export enum TradingView {
-    ADVANCED = 'advanced',
-    TARGET = 'target',
-    SUPPORT = 'support',
-    HEDGE = 'hedge',
-    PORTFOLIO = 'portfolio'
-}
-
-export enum Interval {
-    DAILY = "daily"
+  ADVANCED = "advanced",
+  TARGET = "target",
+  SUPPORT = "support",
+  HEDGE = "hedge",
+  PORTFOLIO = "portfolio"
 }
 
 /**************************************
  *          ARROW INTERFACES          *
  **************************************/
 
-export interface Greeks {
-    delta: number; // Sensitivity of an option’s price to changes in the value of the underlying.
-    gamma: number; // Change in delta per change in price of the underlying.
-    rho: number; // Sensitivity of option prices to changes in interest rates.
-    theta: number; // Measures time decay of price of option.
-    vega: number; // Change in value from a 1% change in volatility.
-}
-
-export interface OptionContract {
-    ticker: Ticker; // Ticker enum that denotes a particular asset.
-    expiration: string; // Readable expiration date in "MMDDYYYY" format (e.g. "01252022" for January 25th, 2022).
-    strike: number[]; // Accepts arrays with two values for spreads. Formatted as [longStrike, shortStrike].
-    contractType: ContractType; // ContractType enum that indicates whether the option is a call, put, call spread, or put spread.
-    price?: number; // Float number that indicates the price of 1 option.
-    spotPrice?: number; // Most up-to-date price of underlying asset.
-    priceHistory?: {
-        date: number;
-        price: number;
-    }[]; // Prices of underlying asset over some period of history.
-    greeks?: Greeks; // Greeks interface that specifies which greeks are tied to this option.
+export interface OptionOrderParams extends Position {
+  payPremium?: boolean // Set to `True` if the user will pay the premium using stablecoin. Set to `False` to pay the premium using their collateral
+  readableExpiration: string // The readable expiration of the option
+  orderType: OrderType // OrderType enum that indicates whether this option is a long open, long close, short open, or short close.
+  thresholdPrice?: number // The minimum (or maximum) price the user is willing to receive (or pay) for this specific option.
 }
 
-export interface OptionOrderParams extends OptionContract {
-    payPremium?: boolean, // Set to `True` if the user will pay the premium using stablecoin. Set to `False` to pay the premium using their collateral
-    quantity?: number; // Float number of contracts desired in the order.
-    orderType: OrderType // OrderType enum that indicates whether this option is a long open, long close, short open, or short close.
-    thresholdPrice?: number; // The minimum (or maximum) price the user is willing to receive (or pay) for this specific option.
+export interface DeliverOptionParams extends OptionOrderParams {
+  expiration: any
```

Comment:
Why are these three entries all `any`? Is it possible for us to be a bit more explicit?